opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/JFlexTokenizer.java

```diff
@@ -119,18 +128,44 @@ public void sourceCodeSeen(SourceCodeSeenEvent evt) {
 
     /**
      * Clears, and then resets the instances attributes per the specified
-     * arguments.
+     * arguments. If {@code start} or {@code end} is past
+     * {@link Integer#MAX_VALUE}, then only the clearing occurs.
      * @param str the matched symbol
      * @param start the match start position
      * @param end the match end position
      */
-    protected void setAttribs(String str, int start, int end) {
+    protected void setAttribs(String str, long start, long end) {
+        clearAttributesEtc();
+        if (start < Integer.MAX_VALUE && end < Integer.MAX_VALUE) {
+
+            if (str.length() > LUCENE_MAX_TOKEN_LENGTH) {
+                str = str.substring(0, LUCENE_MAX_TOKEN_LENGTH);
+                /*
+                 * Leave `end` unadjusted. The truncated string will represent
+                 * the full source text, similar to how a Lucene synonym is an
+                 * alternative representation of full source text.
+                 */
+            }
+
+            /*
+             * FIXME increasing below by one(default) might be tricky, need more
+             * analysis after lucene upgrade to 3.5 below is most probably not
+             * even needed.
+             */
+            this.posIncrAtt.setPositionIncrement(1);
+            this.termAtt.setEmpty();
+            this.termAtt.append(str);
+            this.offsetAtt.setOffset((int) start, (int) end);
+            this.didSetAttribsValues = true;
+        }
+    }
+
+    /**
+     * Calls {@link #clearAttributes()}, and also resets some additional tracked
+     * state.
+     */
+    protected void clearAttributesEtc() {
         clearAttributes();
-        //FIXME increasing below by one(default) might be tricky, need more analysis
-        // after lucene upgrade to 3.5 below is most probably not even needed        
-        this.posIncrAtt.setPositionIncrement(1);
-        this.termAtt.setEmpty();
-        this.termAtt.append(str);
-        this.offsetAtt.setOffset(start, end);
+        didSetAttribsValues = false;
```

Comment:
nit/question: why not use `this.didSetAttribsValues` like in `setAttribs()` ?

Reply:
Oh I preserved the former syntax by the previous author. In methods I don't normally use `this.` unless disambiguation is needed — e.g. often within setters. In the new `clearAttributesEtc()` there was no ambiguity.